sumologic/resource_sumologic_generic_polling_source.go

```diff
@@ -278,29 +293,35 @@ func getPollingTagFilters(d *schema.ResourceData) []TagFilter {
 	return filters
 }
 
-func getPollingAuthentication(d *schema.ResourceData) PollingAuthentication {
+func getPollingAuthentication(d *schema.ResourceData) (PollingAuthentication, error) {
 	auths := d.Get("authentication").([]interface{})
 	authSettings := PollingAuthentication{}
 
 	if len(auths) > 0 {
 		auth := auths[0].(map[string]interface{})
 		switch authType := auth["type"].(string); authType {
 		case "S3BucketAuthentication":
+			if d.Get("content_type").(string) == "AwsInventory" {
+				return authSettings, errors.New(
+					fmt.Sprintf("[ERROR] Unsupported authType: %v for AwsInventory source", authType))
```

Comment:
do we block source creation on the backend when the auth type is bucket auth for inventory sources? if so, I'm wondering if we need to repeat the validation in the tf layer

Reply:
We don't actually. We are, however, considering adding support in the backend, so if and when that is done, I will remove this check from here. OTOH, if we end up deciding not to support bucket auth in backend, we will make that check in the backend in addition to this one.